File: src/entity/board.ts
import type { MarkState } from './idiom';

export interface PositionMarksRaw {
  char: string;
  sm: string;
  ym: string;
  tone: string;
}

export interface ParsedCell {
  col: number;
  attr: string;
  text: string;
  rgb: number[];
  mark: string;
}

export interface ParsedGuess {
  word: string;
  marks: PositionMarksRaw[];
  cells?: ParsedCell[];
}

export interface BoardParseResult {
  guesses: ParsedGuess[];
  warnings: string[];
  /** 模型原始返回文本 */
  raw_response?: string;
}

export type OcrMarkState = Exclude<MarkState, null>;

export interface OcrReviewRow {
  word: string;
  marks: {
    char: OcrMarkState;
    sm: OcrMarkState;
    ym: OcrMarkState;
    tone: OcrMarkState;
  }[];
  selected: boolean;
  missingLexicon: boolean;
}

export function rawMarkToState(raw: string): OcrMarkState {
  if (raw === 'hit' || raw === 'present' || raw === 'absent') return raw;
  return 'absent';
}

File: src-tauri/src/entity/board.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PositionMarksRaw {
    pub char: String,
    pub sm: String,
    pub ym: String,
    pub tone: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ParsedCell {
    pub col: usize,
    pub attr: String,
    pub text: String,
    pub rgb: Vec<u8>,
    pub mark: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ParsedGuess {
    pub word: String,
    pub marks: Vec<PositionMarksRaw>,
    #[serde(default)]
    pub cells: Vec<ParsedCell>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct BoardParseResult {
    pub guesses: Vec<ParsedGuess>,
    #[serde(default)]
    pub warnings: Vec<String>,
    /// 模型原始返回文本（便于排查与历史记录）
    #[serde(default)]
    pub raw_response: String,
}
